portlist/portlist_linux.go

```diff
@@ -80,16 +89,22 @@ func listPorts() (List, error) {
 			rem := words[2]
 			inode := words[9]
 
-			// If a port is bound to 127.0.0.1, ignore it.
-			if strings.HasPrefix(local, "0100007F:") {
+			// If a port is bound to localhost, ignore it.
+			// TODO: localhost is bigger than 1 IP, we need to ignore
+			// more things.
+			if strings.HasPrefix(local, localhost) {
 				continue
 			}
-			if rem != "00000000:0000" {
+			if rem != any {
 				// not a "listener" port
 				continue
 			}
 
-			portv, err := strconv.ParseUint(local[9:], 16, 16)
+			localPort := strings.Split(local, ":")
```

Comment:
this allocates. This code used to show up in profiles, which was what motivated 120273d7f68bda9b54ee30803fb1067c1f26ca7b so I'd like not regress too much if it's not too hard.